addons/l10n_eg_edi_eta/views/account_move_view.xml

```diff
@@ -0,0 +1,83 @@
+<odoo>
+    <data>
+        <record id="action_sign_invoices" model="ir.actions.server">
+            <field name="name">Sign invoices</field>
+            <field name="type">ir.actions.server</field>
+            <field name="state">code</field>
+            <field name="model_id" ref="account.model_account_move"/>
+            <field name="binding_model_id" ref="account.model_account_move"/>
+            <field name="code">
+                action = records.action_post_sign_invoice()
+            </field>
+        </record>
+
+
+        <record id="view_move_form_inherit" model="ir.ui.view">
+            <field name="name">view_move_form_inherit</field>
+            <field name="model">account.move</field>
+            <field name="inherit_id" ref="account.view_move_form"/>
+            <field name="priority" eval="40"/>
+            <field name="arch" type="xml">
+                <xpath expr="//header/button[@name='action_post']" position="after">
+                    <button name="action_post_sign_invoice" type="object"
+                            class="oe_highlight"
+                            groups="account.group_account_invoice"
+                            string="Sign Invoice"
+                            attrs="{'invisible': ['|', '|', ('country_code', '!=', 'EG'), ('l10n_eg_signature_data', '=', True), ('state', '=', 'draft')]}"/>
+                </xpath>
+                <xpath expr="//field[@name='invoice_date']" position="after">
+                    <field name="l10n_eg_posted_datetime" readonly="1" attrs="{'invisible': [('country_code', '!=', 'EG')]}"/>
+                </xpath>
+                <notebook position="inside">
+                    <page string="ETA E-Invoice" attrs="{'invisible': [('country_code', '!=', 'EG')]}">
+                        <group>
+                            <group>
+                                <field name="l10n_eg_uuid" readonly="1" force_save="1"/>
+                                <field name="l10n_eg_submission_id" readonly="1" force_save="1"/>
+                                
+                            </group>
+                            <group>
+                                <field name="l10n_eg_signature_data" readonly="1" invisible='1'/>
+                            </group>
+                        </group>
+                        <group>
+                            <group attrs="{'invisible': [('edi_state','!=', 'sent')]}">
+                                <label for='l10n_eg_pdf'/>
+                                <div class="o_row">
+                                    <field name="l10n_eg_pdf" filename="eta_document_name" readonly="1" force_save="1"/>
```

Comment:
Maybe we should just put this in the chatter by using many2one to ir.attachment and giving it as res_id and res_model.  We do not put attachments in the view like that mostly.  This will also be weird in the portal as it will not be seen as the official attachment probably.